corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,416 @@
+
+    public static void main(String[] args) throws Exception {
+        CorfuStoreCompactorMain corfuCompactorMain = new CorfuStoreCompactorMain();
+        corfuCompactorMain.getCompactorArgs(args);
+
+        CorfuRuntimeHelper corfuRuntimeHelper;
+        CorfuRuntimeHelper cpRuntimeHelper;
+        if (maxWriteSize == -1) {
+            if (persistedCacheRoot == null) {
+                // in-memory compaction
+                maxWriteSize = DEFAULT_CP_MAX_WRITE_SIZE;
+            } else {
+                // disk-backed non-config compaction
+                maxWriteSize = NON_CONFIG_DEFAULT_CP_MAX_WRITE_SIZE;
+            }
+        }
+        if (tlsEnabled) {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+
+        } else {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+        }
+
+        corfuRuntime = corfuRuntimeHelper.getRuntime();
+        corfuStore = new CorfuStore(corfuRuntime);
+        corfuCompactor = new CorfuStoreCompactor(corfuRuntime, cpRuntimeHelper.getRuntime(), trim, persistedCacheRoot, isLeader);
+        thisNodeUuid = UUID.fromString(CorfuRuntimeHelper.getThisNodeUuid());
+
+        openCompactionTables();
+
+        if (isCheckpointFrozen()) {
+            return;
+        }
+
+        if (trim) {
+            trimAndUpdateToken();
+        }
+
+        //TODO: Write a plugin for upgrade?
+        if (isUpgrade) {
+            log.info("Upgrade: Saving Trim Token");
+
+            if (upgradeDescriptorTable) {
+                syncProtobufDescriptorTable();
+            }
+            final Optional<TokenMsg> minToken = getGlobalToken();
+            if (minToken.isPresent()) {
+                log.info("Upgrade: Saving Trim Token {}", minToken.get());
+                FileWriter fileWriter = new FileWriter(trimTokenFile, true);
+                PrintWriter printWriter = new PrintWriter(fileWriter);
+                printWriter.println(minToken.get());
+                printWriter.close();
+            } else {
+                log.warn("Upgrade: Trying to save trim token, but got null minToken!");
+            }
+        }
+
+        try {
+            corfuCompactor.checkpoint();
+        } catch (Throwable throwable) {
+            log.warn("CorfuStoreCompactorMain crashed with error:", throwable);
+            //TODO: Find a way to log this error into syslog..
```

Comment:
get rid of todo

Reply:
This was a huge change but finally we can remove the TODO given that full syslog integration is complete.
Just a note for future though, these types of integrations may not always be possible in the scope of the same PR, so in my opinion sometimes placing TODOs should not be considered an error.